### PR TITLE
🕷️ #2553 Updated CoreCompletion Calculation

### PR DESCRIPTION
**Pull request by demariadaniel**
**Description of changes**

Updates Specimen Core Completion Calculation
- Only Specimens w/ DNA Sample Registrations are counted
- Must include Specimen Clinical Info (specimen has been submitted) and calculates ratio of registered Samples to incomplete Specimens when clinical info is missing (and not 0)
- Must have at least 1 Normal / Tumour record for completion

Related Tickets:
UI #2580
https://app.zenhub.com/workspaces/icgc-argo-platform-dk-production-board-5e542d38415f5034e9fed89d/issues/gh/icgc-argo/platform-ui/2580

UI #2553
https://app.zenhub.com/workspaces/icgc-argo-platform-dk-production-board-5e542d38415f5034e9fed89d/issues/gh/icgc-argo/platform-ui/2553

**Type of Change**

- [x] Bug
- [x] Refactor
- [ ] New Feature
- [ ] Release Candidate

**Checklist before requesting review:**

- [ ] Check branch (code change PRs go to `develop` not master)
- [ ] Check copyrights for new files
- [ ] Manual testing
- [ ] Regression tests completed and passing (double check number of tests).
- [ ] Spelling has been checked.
- [ ] Updated swagger docs accordingly (check it's still valid)
- [ ] Set `validationDependency` in meta tag for [Argo Dictionary](https://github.com/icgc-argo/argo-dictionary) fields used in code

### CODE REVIEW

File: src/common-model/functions.ts [hunk 1]
```diff
@@ -18,16 +18,22 @@
  */
 
 import { DeepReadonly } from 'deep-freeze';
-import { Donor, ClinicalEntity, ClinicalInfo, Treatment } from '../clinical/clinical-entities';
+import _, { isEmpty } from 'lodash';
```
Comment (ciaranschutte) [inline src/common-model/functions.ts:21]: we can remove `_` default import and keep using the named exports

File: src/common-model/functions.ts [hunk 2]
```diff
@@ -218,6 +224,39 @@ export function getSampleRegistrationDataFromDonor(donor: DeepReadonly<Donor>) {
   return sample_registration;
 }
 
+export const dnaSampleFilter = (specimen: Specimen): boolean =>
+  // All Specimen Have a Sample Registration
+  // Only DNA records count towards completion
+  specimen.samples.filter(sample => dnaSampleTypes.includes(sample.sampleType)).length > 0;
```
Comment (ciaranschutte) [inline src/common-model/functions.ts:230]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some
work here?

Comment (demariadaniel) [reply src/common-model/functions.ts]: Yes it does

File: src/common-model/functions.ts (hunk 1 above)
Comment (demariadaniel) [reply src/common-model/functions.ts]: Updated

**Review by Buwujiu — APPROVED** [review]

**Pull request merged**